quickwit-config/src/server_config.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::ffi::OsStr;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Context};
+use byte_unit::Byte;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct IndexerConfig {
+    pub data_dir_path: PathBuf,
+}
+
+impl IndexerConfig {
+    pub fn validate(&self) -> anyhow::Result<()> {
+        if !self.data_dir_path.exists() {
+            bail!(
+                "Data dir `{}` does not exist.",
+                self.data_dir_path.display()
+            );
+        }
+        Ok(())
+    }
+}
+
+// TODO: caching
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SearcherConfig {
+    pub data_dir_path: PathBuf,
+    pub peer_seeds: Vec<String>,
```

Comment:
why not `SocketAddr` instead of `String` for peer ?

Reply:
`SocketAddr` only accepts IPs but we want to be able to provide hostnames.